src/printer/insert-new-lines.ts

```diff
@@ -330,35 +331,35 @@ function insertLinesIntoArray(
                                     parentToMutate[siblingIndex] = commaSibling;
                                 });
                             }
-                        } else if (isDocCommand(currentDoc) && currentDoc.type === 'concat') {
-                            const firstPart = currentDoc.parts[0];
-                            const secondPart = currentDoc.parts[1];
-                            if (debug) {
-                                console.info('got concat child doc');
-                                console.info(currentDoc.parts, {firstPart, secondPart});
-                                console.info(
-                                    isDocCommand(firstPart),
-                                    isDocCommand(secondPart),
-                                    (firstPart as any).type === 'line',
-                                    (firstPart as any).hard,
-                                    (secondPart as any).type === 'break-parent',
-                                );
-                            }
-                            if (
-                                isDocCommand(firstPart) &&
-                                isDocCommand(secondPart) &&
-                                firstPart.type === 'line' &&
-                                firstPart.hard &&
-                                secondPart.type === 'break-parent'
-                            ) {
-                                if (debug) {
-                                    console.info('concat child was indeed a line break');
-                                }
-                                forceFinalLineBreakExists = true;
-                                return false;
-                            } else if (debug) {
-                                console.info('concat child doc was not a line break');
-                            }
+                        // } else if (isDocCommand(currentDoc) && currentDoc.type === 'concat') {
```

Comment:
Where `currentDoc.type === 'concat'` was true in Prettier v2, in v3 it's just an array (according to the changelog docs [here](https://prettier.io/blog/2023/07/05/3.0.0.html#update-prettierdoc-13203httpsgithubcomprettierprettierpull13203-14456httpsgithubcomprettierprettierpull14456-by-fiskerhttpsgithubcomfisker) and [here](https://prettier.io/blog/2021/05/09/2.3.0#use-arrays-instead-of-concat-9733httpsgithubcomprettierprettierpull9733-by-fiskerhttpsgithubcomfisker-thorn0httpsgithubcomthorn0))

Reply:
[richard.patch](https://github.com/electrovir/prettier-plugin-multiline-arrays/files/12560269/richard.patch)

My understanding is that this should fix it, but the tests fail because of the other issue so I'm not really able to validate if it's actually correct. This is based on my understanding of the docs, I have no previous experience in building Prettier plugins 😅